test/performance/async_perf_test.py

```diff
@@ -0,0 +1,456 @@
+
+    async def after(self):
+        await self.client.perftest.drop_collection("corpus")
+
+
+class FindTest(TestDocument):
+    dataset = "tweet.json"
+
+    async def asyncSetUp(self):
+        await super().asyncSetUp()
+        self.data_size = len(encode(self.document)) * NUM_DOCS
+        documents = [self.document.copy() for _ in range(NUM_DOCS)]
+        self.corpus = self.client.perftest.corpus
+        result = await self.corpus.insert_many(documents)
+        self.inserted_ids = result.inserted_ids
+
+    async def before(self):
+        pass
+
+    async def after(self):
+        pass
+
+
+class TestFindOneByID(FindTest, AsyncPyMongoTestCase):
+    async def do_task(self):
+        find_one = self.corpus.find_one
+        for _id in self.inserted_ids:
+            await find_one({"_id": _id})
+
+
+class TestFindOneByID8Tasks(TestFindOneByID):
+    n_tasks = 8
+
+
+class TestFindOneByID80Tasks(TestFindOneByID):
+    n_tasks = 80
+
+
+class TestFindOneByIDManyTasks(TestFindOneByID):
+    async def do_task(self):
+        find_one = self.corpus.find_one
+        await asyncio.gather(*[find_one({"_id": _id}) for _id in self.inserted_ids])
+
+
+class SmallDocInsertTest(TestDocument):
+    dataset = "small_doc.json"
+
+    async def asyncSetUp(self):
+        await super().asyncSetUp()
+        self.data_size = len(encode(self.document)) * NUM_DOCS
+        self.documents = [self.document.copy() for _ in range(NUM_DOCS)]
+
+
+class SmallDocMixedTest(TestDocument):
+    dataset = "small_doc.json"
+
+    async def asyncSetUp(self):
+        await super().asyncSetUp()
+        self.data_size = len(encode(self.document)) * NUM_DOCS * 2
+        self.documents = [self.document.copy() for _ in range(NUM_DOCS)]
+
+
+class TestSmallDocInsertOne(SmallDocInsertTest, AsyncPyMongoTestCase):
+    async def do_task(self):
+        insert_one = self.corpus.insert_one
+        await asyncio.gather(*[insert_one(doc) for doc in self.documents])
+
+
+class LargeDocInsertTest(TestDocument):
+    dataset = "large_doc.json"
+
+    async def asyncSetUp(self):
+        await super().asyncSetUp()
+        n_docs = 10
+        self.data_size = len(encode(self.document)) * n_docs
+        self.documents = [self.document.copy() for _ in range(n_docs)]
+
+
+class TestLargeDocInsertOne(LargeDocInsertTest, AsyncPyMongoTestCase):
+    async def do_task(self):
+        insert_one = self.corpus.insert_one
+        await asyncio.gather(*[insert_one(doc) for doc in self.documents])
+
+
+# MULTI-DOC BENCHMARKS
+class TestFindManyAndEmptyCursor(FindTest, AsyncPyMongoTestCase):
+    async def do_task(self):
+        await self.corpus.find().to_list()
+
+
+class TestFindManyAndEmptyCursor8Tasks(TestFindManyAndEmptyCursor):
+    n_tasks = 8
+
+
+class TestFindManyAndEmptyCursor80Tasks(TestFindManyAndEmptyCursor):
+    n_tasks = 80
+
+
+class TestFindManyAndEmptyCursorManyTasks(TestFindManyAndEmptyCursor):
+    n_tasks = 1000
```

Comment:
Should we remove the n_tasks=1000 benchmarks? We already have 80, do we need 1000?